base/errorshow.jl

```diff
@@ -139,7 +139,7 @@ function showerror(io::IO, ex::TypeError)
         elseif isa(ex.got, Type)
             targs = ("Type{", ex.got, "}")
         else
-            targs = (typeof(ex.got),)
+            targs = ("$(ex.got) instance of $(typeof(ex.got))",)
```

Comment:
```suggestion
            targs = ("$(ex.got) with type $(typeof(ex.got))",)
```

Reply:
```suggestion
            targs = ("a $(typeof(ex.got))",)
```

Reply:
How about "got a value of type $T"? Avoids slightly awkward things like "a Int".

Reply:
sure! updated in bca5f7a